apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,408 @@
+        /// <param name="lhs">The value to compare with <paramref name="lhs" />.</param>
+        /// <param name="rhs">The value to compare with <paramref name="rhs" />.</param>
+        /// <returns><c>true</c> if <paramref name="lhs" /> is greater than <paramref name="rhs" />; otherwise, <c>false</c>.</returns>
+        public static bool operator >(SpannerDate lhs, SpannerDate rhs) => lhs.CompareTo(rhs) > 0;
+
+        /// <summary>Compares two values to determine which is greater or equal.</summary>
+        /// <param name="lhs">The value to compare with <paramref name="lhs" />.</param>
+        /// <param name="rhs">The value to compare with <paramref name="rhs" />.</param>
+        /// <returns><c>true</c> if <paramref name="lhs" /> is greater than or equal to <paramref name="rhs" />; otherwise, <c>false</c>.</returns>
+        public static bool operator >=(SpannerDate lhs, SpannerDate rhs) => lhs.CompareTo(rhs) >= 0;
+
+        /// <summary>Adds days to the SpannerDate to compute the new SpannerDate.</summary>
+        /// <param name="date">The value to which <paramref name="days" /> is added.</param>
+        /// <param name="days">The value which is added to <paramref name="date" />.</param>
+        /// <returns>The new SpannerDate obtained by adding <paramref name="days" /> to <paramref name="date" />.</returns>
+        public static SpannerDate operator +(SpannerDate date, int days) => new SpannerDate(date._days + days);
+
+        /// <summary>Subtracts the days from the SpannerDate.</summary>
+        /// <param name="date">The SpannerDate from which <paramref name="days" /> is subtracted.</param>
+        /// <param name="days">The value which is subtracted from the <paramref name="date" />.</param>
+        /// <returns>The difference of <paramref name="days" /> subtracted from <paramref name="date" />.</returns>
+        public static SpannerDate operator -(SpannerDate date, int days) => date + -days;
+
+        /// <summary>Subtracts two values to compute their differences.</summary>
+        /// <param name="lhs">The SpannerDate from which <paramref name="rhs" /> is subtracted.</param>
+        /// <param name="rhs">The SpannerDate which is subtracted from the <paramref name="lhs" />.</param>
+        /// <returns>The difference of <paramref name="rhs" /> subtracted from <paramref name="lhs" />.</returns>
+        public static int operator -(SpannerDate lhs, SpannerDate rhs) => lhs._days - rhs._days;
+
+        /// <summary>
+        /// Performs an implicit conversion from <see cref="DateTime"/> to <see cref="SpannerDate"/>.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <returns>
+        /// The result of the conversion.
+        /// </returns>
+        public static implicit operator SpannerDate(DateTime value) => new SpannerDate(value);
+
+        /// <summary>
+        /// Performs an explicit conversion from <see cref="SpannerDate"/> to <see cref="DateTime"/>.
```

Comment:
Any reason for this being explicit while the reverse is implicit? I'd argue that if anything, they should be the other way round, given that the conversion to SpannerDate loses information (time) whereas the conversion to DateTime doesn't. I think it's probably clearer if both are explicit though.